core/src/main/java/org/apache/iceberg/SingleValueParser.java

```diff
@@ -309,17 +309,28 @@ public static void toJson(Type type, Object defaultValue, JsonGenerator generato
         generator.writeString(defaultValue.toString());
         break;
       case FIXED:
+        // Normally, FIXED is backed by a byte[], but it can also be backed by a ByteBuffer in the
+        // case of Fixed Literals. Ideally, Fixed Literals would be backed by a byte[], but that
+        // would make the APIs backwards incompatible.
+        // See {@link org.apache.iceberg.expressions.Literals.FixedLiteral}.
         Preconditions.checkArgument(
-            defaultValue instanceof ByteBuffer, "Invalid default %s value: %s", type, defaultValue);
-        ByteBuffer byteBufferValue = (ByteBuffer) defaultValue;
```

Comment:
Can you revert these changes so I can see the failure?

Reply:
No longer applicable after fixing the issue as per the above discussion.